walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,155 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
 
 public final class FindMeetingQuery {
+  /**
+   * Gets the available time slot for new meeting given existing meeting events.
+   * @param events Existing meeting events.
+   * @param request Request to set up a new meeting
+   * @return If one or more time slots exists so that both mandatory and optional attendees can attend,
+   *         returns those time slots. Otherwise, returns the time slots that fit just the mandatory attendees.
+   *         Returns empty if duration is greater than a day.
+   */
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    if (events.isEmpty()) {
```

Comment:
Why do we need a repeated check here for empty event, wouldn't empty event make occupied time range empty and the same thing returned?
Besides, I think if you has some public facing parameter requirement, you can check early here instead of letting them propagate to the other private methods. 
E.g param request need to have 
+ valid duration: 0 < request.duration <= WHOLE_DAY.
+ valid attendee: >= 1

Invalid request get empty response. Make immutable as well.
Empty event list is acceptable I think.

Inside getAvailableTimeRange you can check empty occupied time and return whole day?

Do we accept request that has duration  > one day?